salt/pillar/rethinkdb_pillar.py

```diff
@@ -0,0 +1,159 @@
+# -*- coding: utf-8 -*-
+'''
+Provide external pillar data from RethinkDB
+
+:depends: rethinkdb (on the salt-master)
+
+
+salt master rethinkdb configuration
+===================================
+These variables must be configured in your master configuration file.
+    * ``rethinkdb.host`` - The RethinkDB server. Defaults to ``'salt'``
+    * ``rethinkdb.port`` - The port the RethinkDB server listens on.
+      Defaults to ``'28015'``
+    * ``rethinkdb.database`` - The database to connect to.
+      Defaults to ``'salt'``
+    * ``rethinkdb.username`` - The username for connecting to RethinkDB.
+      Defaults to ``''``
+    * ``rethinkdb.password`` - The password for connecting to RethinkDB.
+      Defaults to ``''``
+
+
+salt-master ext_pillar configuration
+====================================
+
+The ext_pillar function arguments are given in single line dictionary notation.
+
+.. code-block:: yaml
+
+  ext_pillar:
+    - rethinkdb: {table: ext_pillar, id_field: minion_id, field: pillar_root, pillar_key: external_pillar}
+
+In the example above the following happens.
+    * The salt-master will look for external pillars in the 'ext_pillar' table
+      on the RethinkDB host
+    * The minion id will be matched against the 'minion_id' field
+    * Pillars will be retrieved from the nested field 'pillar_root'
+    * Found pillars will be merged inside a key called 'external_pillar'
+
+
+Module Documentation
+====================
```

Comment:
Did you intend to add some documentation here? If not, then I think this section header should be removed.

Reply:
I took some inspiration from the mongodb external pillar module, where I liked how the function docstrings have a section header above them which makes it look more organized on the docs page. If it is bad practice, then I don't mind removing it.

Reply:
Ah, ok. I think that is fine. I was just confused why there weren't docs there, but that makes sense when the docs are built out.